folium/plugins/timedynamic_geo_json.py

```diff
@@ -0,0 +1,175 @@
+"""
+module containing the class TimeDynamicGeoJson, which can be used to
+create choropleth maps with a time slider.
+"""
+from jinja2 import Template
+from branca.element import JavascriptLink, Figure
+from folium.features import GeoJson
+
+
+class TimeDynamicGeoJson(GeoJson):
+    """
+    Creates a GeoJson object for plotting into a Map.
+
+    Parameters
+    ----------
+    data: file, dict or str.
+        The GeoJSON data you want to plot.
+        * If file, then data will be read in the file and fully
+        embedded in Leaflet's JavaScript.
+        * If dict, then data will be converted to JSON and embedded
+        in the JavaScript.
+        * If str, then data will be passed to the JavaScript as-is.
+    style_function: function, default None
+        A function mapping a GeoJson Feature to a style dict.
+    name : string, default None
+        The name of the Layer, as it will appear in LayerControls
+    overlay : bool, default False
+        Adds the layer as an optional overlay (True) or the base layer (False).
+    control : bool, default True
+        Whether the Layer will be included in LayerControls
+    smooth_factor: float, default None
+        How much to simplify the polyline on each zoom level. More means
+        better performance and smoother look, and less means more accurate
+        representation. Leaflet defaults to 1.0.
```

Comment:
You can probably remove all that, refer to the `GeoJson` docs, and add only what is different about this `plugin`.